### PR TITLE
add support for SymNodeVariable in getitem_const

### PR DESCRIPTION
**Pull request by blzheng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97756

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -7008,6 +7008,24 @@ def fn(a):
             with self.assertRaises(RuntimeError):
                 torch.compile(fn)(a)
 
+        def test_getitem(self):
```
Comment (jgong5) [inline test/inductor/test_torchinductor.py:7011]: Is it common for both CPU and GPU? If it is common, can you move it outside of CPU repro?

Comment (blzheng) [reply test/inductor/test_torchinductor.py]: Done.

**Review by jgong5 — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

File: torch/_dynamo/variables/lists.py [hunk 2]
```diff
@@ -53,7 +53,13 @@ def as_proxy(self):
         return self.python_type()(self._as_proxy())
 
     def getitem_const(self, arg: VariableTracker):
-        index = arg.as_python_constant()
+        from .tensor import SymNodeVariable
+
+        if isinstance(arg, SymNodeVariable):
+            index = arg.sym_num
```
Comment (jansel) [inline torch/_dynamo/variables/lists.py:59]: @ezyang is this right way to trigger a guard_as_constant on a symint?

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**